dali/operators/input/video_input.cc

```diff
@@ -0,0 +1,166 @@
+``last_sequence_policy`` parameter. For example::
+
+
+    This is a video that consists of 67 frames (every '-' is a frame):
+    -------------------------------------------------------------------
+
+
+    User decided that there shall be 5 frames per sequence and the last_sequence_policy='partial':
+    -------------------------------------------------------------------
+    [   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][]
+    -------------------------------------------------------------------
+                      Since there are not enough frames, the last sequence comprises 2 frames.
+
+
+    The Pipeline has max_batch_size=3, therefore the operator will return 5 batches of sequences.
+    First 4 batches comprise 3 sequences and the last batch is partial and comprises 2 sequences.
+    ---------------   ---------------   ---------------   ---------------   -------
+    [   ][   ][   ]   [   ][   ][   ]   [   ][   ][   ]   [   ][   ][   ]   [   ][]
+    ---------------   ---------------   ---------------   ---------------   -------
+
+
+    With the last_sequence_policy='pad', the last sequence of the last batch will be padded with 0:
+    ---------------   ---------------   ---------------   ---------------   -------000
+    [   ][   ][   ]   [   ][   ][   ]   [   ][   ][   ]   [   ][   ][   ]   [   ][   ]
+    ---------------   ---------------   ---------------   ---------------   -------000
+
+
+The difference between ``fn.inputs.video`` and ``fn.readers.video`` is that the former
+reads an encoded video from memory and the latter reads the encoded video from disk.
+
+The difference between ``fn.inputs.video`` and ``fn.decoders.video`` is that the former
+does not decode the whole video file in one go. This behaviour is needed for longer videos. E.g.
+5-min, 4k, 30fps decoded video takes about 1.7 TB of memory.
+
+This operator accepts most of the video containers and file formats. FFmpeg is used to parse
+the video container. In the situations, that the container does not contain required metadata
+(e.g. frames sizes, number of frames, etc...), the operator needs to find it out itself,
+which may result in a slowdown.
+)code")
+                .NumInput(0)
+                .NumOutput(1)
+                .AddArg("sequence_length", R"code(
+Number of frames in each sequence.
+)code", DALI_INT32)
+                .AddOptionalArg("last_sequence_policy", R"code(
+Specifies, how to handle the last sequence in the video file.
+
+For a given number of frames in the video file and ``frames_per_sequence`` parameter,
+it might happen that the video can't be split uniformly across sequences. If the
+``last_sequence_policy='partial'``, the last sequence might have fewer frames than
+``frames_per_sequence`` value specified. If the ``last_sequence_policy='partial'``,
+the last sequence will always have ``frames_per_sequence`` frames and will
+be padded with empty frames.
```

Comment:
```suggestion
For a given number of frames in the video file and ``sequence_length`` parameter,
it might happen that the video can't be split uniformly across sequences. If the
``last_sequence_policy='partial'``, the last sequence might have fewer frames than
``sequence_length`` value specified. If the ``last_sequence_policy='partial'``,
the last sequence will always have ``sequence_length`` frames and will
be padded with empty frames.
```

Leftover.